### PR TITLE
rspconfig dump to allow admins capture logs

### PR DESCRIPTION
**Pull request by xuweibj**
#4331 

Output of ``rspconfig dump``
```
# rspconfig mid05tor12cn11,mid05tor12cn16 dump
mid05tor12cn16: Capturing BMC Dump information, this will take some time...
mid05tor12cn11: Capturing BMC Dump information, this will take some time...
mid05tor12cn16: Generated BMC Dump 40
mid05tor12cn11: Generated BMC Dump 33
mid05tor12cn16: Downloaded Dump 40 to /var/log/xcat/dump/mid05tor12cn16_dump_40.tar.xz
mid05tor12cn11: Downloaded Dump 33 to /var/log/xcat/dump/mid05tor12cn11_dump_33.tar.xz
```

Output of ``rspconfig dump -V``
```
# rspconfig mid05tor12cn11,mid05tor12cn16 dump -V
mid05tor12cn16: [briggs01]: Capturing BMC Dump information, this will take some time...
mid05tor12cn11: [briggs01]: Capturing BMC Dump information, this will take some time...
mid05tor12cn16: [briggs01]: Generating BMC Dump...
mid05tor12cn11: [briggs01]: Generating BMC Dump...
mid05tor12cn11: [briggs01]: Generating BMC Dump...
mid05tor12cn16: [briggs01]: Generating BMC Dump...
mid05tor12cn16: [briggs01]: Generated BMC Dump 41
mid05tor12cn16: [briggs01]: Downloading Dump...
mid05tor12cn11: [briggs01]: Generated BMC Dump 34
mid05tor12cn11: [briggs01]: Downloading Dump...
mid05tor12cn16: [briggs01]: Downloaded Dump 41 to /var/log/xcat/dump/mid05tor12cn16_dump_41.tar.xz
mid05tor12cn11: [briggs01]: Downloaded Dump 34 to /var/log/xcat/dump/mid05tor12cn11_dump_34.tar.xz
```

### CODE REVIEW

File: xCAT-server/lib/xcat/plugins/openbmc.pm [hunk 1]
```diff
@@ -1183,28 +1191,36 @@ sub parse_command_status {
...
-                $next_status{LOGIN_RESPONSE} = "RSPCONFIG_DUMPDWLD_REQUEST";
-                $next_status{RSPCONFIG_DUMPDWLD_REQUEST} = "RSPCONFIG_DUMPDWLD_RESPONSE";
-                $status_info{RSPCONFIG_DUMPDWLD_REQUEST}{init_url} =~ s/#ID#/$$subcommands[2]/g; 
-                $status_info{RSPCONFIG_DUMPDWLD_REQUEST}{argv} = $$subcommands[2];
+                $next_status{LOGIN_RESPONSE} = "RSPCONFIG_DUMP_DOWNLOAD_REQUEST";
+                $next_status{RSPCONFIG_DUMP_DOWNLOAD_REQUEST} = "RSPCONFIG_DUMP_DOWNLOAD_RESPONSE";
+                $status_info{RSPCONFIG_DUMP_DOWNLOAD_REQUEST}{init_url} =~ s/#ID#/$$subcommands[2]/g; 
+                $status_info{RSPCONFIG_DUMP_DOWNLOAD_REQUEST}{argv} = $$subcommands[2];
+            } else {
+                $next_status{LOGIN_RESPONSE} = "RSPCONFIG_DUMP_CREATE_REQUEST";
```
Comment (whowutwut) [inline xCAT-server/lib/xcat/plugins/openbmc.pm]: Let's add a comment here, 

```
# this section handles the dump support where no options are given and xCAT will 
# handle the creation, waiting, and download of the dump across a given noderange 
```

**Comment by whowutwut** [comment]
@gurevichmark I'd like to suggest the following changes to the external, you can fix the wording...

non-verbose: 
```
# rspconfig mid05tor12cn11,mid05tor12cn16 dump
Capturing BMC Diagnostic information, this will take some time...
mid05tor12cn16: Dump Requested.  Target ID is 40, waiting for BMC to generate...
mid05tor12cn11: Dump Requested. Target ID is 33, waiting for BMC to generate...
mid05tor12cn16: Generated, downloading to /var/log/xcat/dump/mid05tor12cn16_dump_40.tar.xz
mid05tor12cn11: Generated, downloading to /var/log/xcat/dump/mid05tor12cn11_dump_33.tar.xz
```

Also thinking if we can add a YYYYMMDDHHMM to the front of the name of the dump file to more easily be able to tar up a group of dumps together.

**Comment by gurevichmark** [comment]
Updated output:

* Single node without `-V`:
```
[root@briggs01 xCAT_plugin]# XCATBYPASS=1 rspconfig mid05tor12cn11 dump
Capturing BMC Diagnostic information, this will take some time...
mid05tor12cn11: Dump requested. Target ID is 54, waiting for BMC to generate...
mid05tor12cn11: Dump 54 generated. Downloading to /var/log/xcat/dump/20171128157_mid05tor12cn11_dump_54.tar.xz
```

* Single node with `-V`:
```
[root@briggs01 xCAT_plugin]# XCATBYPASS=1 rspconfig mid05tor12cn11 dump -V
Capturing BMC Diagnostic information, this will take some time...
mid05tor12cn11: Dump requested. Target ID is 56, waiting for BMC to generate...
mid05tor12cn11: Dump 56 generated. Downloading to /var/log/xcat/dump/201711281525_mid05tor12cn11_dump_56.tar.xz
[root@briggs01 xCAT_plugin]# mid05tor12cn11: Downloaded Dump 56 to /var/log/xcat/dump/201711281525_mid05tor12cn11_dump_56.tar.xz

[root@briggs01 xCAT_plugin]#
```

* Range of nodes:
```
[root@briggs01 xCAT_plugin]# XCATBYPASS=1 rspconfig p9up dump
Capturing BMC Diagnostic information, this will take some time...
mid05tor12cn13: Dump requested. Target ID is 14, waiting for BMC to generate...
mid05tor12cn16: Dump requested. Target ID is 58, waiting for BMC to generate...
mid05tor12cn11: Dump requested. Target ID is 57, waiting for BMC to generate...
mid05tor12cn02: Error: BMC returned 200 OK but no ID was returned.  Verify manually on the BMC.
mid05tor12cn15: Error: DBusException('Dump not captured due to a cap.',)
mid05tor12cn18: Error: BMC returned 200 OK but no ID was returned.  Verify manually on the BMC.
mid05tor12cn05: Error: BMC returned 200 OK but no ID was returned.  Verify manually on the BMC.
mid05tor12cn11: Dump 57 generated. Downloading to /var/log/xcat/dump/201711281529_mid05tor12cn11_dump_57.tar.xz
mid05tor12cn13: Dump 14 generated. Downloading to /var/log/xcat/dump/201711281532_mid05tor12cn13_dump_14.tar.xz
[root@briggs01 xCAT_plugin]#
```

**Comment by gurevichmark** [comment]
There are still some issue to address for this PR:
1. When dump is not being generated we wait 10 min and display this message:
```
[root@briggs01 xCAT_plugin]# XCATBYPASS=1 rspconfig mid05tor12cn16 dump
Capturing BMC Diagnostic information, this will take some time...
mid05tor12cn16: Dump requested. Target ID is 59, waiting for BMC to generate...
mid05tor12cn16: Error: Could not find dump 59 after waiting 600 seconds.
[root@briggs01 xCAT_plugin]#
```
But this message does not get displayed when the same node is included in the range:
```
[root@briggs01 xCAT_plugin]# XCATBYPASS=1 rspconfig p9up dump
Capturing BMC Diagnostic information, this will take some time...
mid05tor12cn13: Dump requested. Target ID is 14, waiting for BMC to generate...
mid05tor12cn16: Dump requested. Target ID is 58, waiting for BMC to generate...  <-- never get error message that dump was not generated
mid05tor12cn11: Dump requested. Target ID is 57, waiting for BMC to generate...
mid05tor12cn02: Error: BMC returned 200 OK but no ID was returned.  Verify manually on the BMC.
mid05tor12cn15: Error: DBusException('Dump not captured due to a cap.',)
mid05tor12cn18: Error: BMC returned 200 OK but no ID was returned.  Verify manually on the BMC.
mid05tor12cn05: Error: BMC returned 200 OK but no ID was returned.  Verify manually on the BMC.
mid05tor12cn11: Dump 57 generated. Downloading to /var/log/xcat/dump/201711281529_mid05tor12cn11_dump_57.tar.xz
mid05tor12cn13: Dump 14 generated. Downloading to /var/log/xcat/dump/201711281532_mid05tor12cn13_dump_14.tar.xz
[root@briggs01 xCAT_plugin]#
```

2. 10 min is probably too long to wait for the dump to be generated. Perhaps 5 min is better, and every 2 min message should be displayed that we are still waiting, otherwise user might think the command hung.

**Pull request merged**